src/echoclient/EchoClient.php

```diff
@@ -429,6 +438,48 @@ protected function getHeaders($noCache = false)
         return $headers;
     }
 
+    /**
+     * Sends json encoded events data to echo
+     *
+     * @param string $eventsData The json encoded events data to send
+     * @return bool True if successful, else false
+     */
+    protected function sendJsonEventDataToEcho($eventsData)
+    {
+        $baseUrl = $this->getBaseUrl();
+
+        if (!$baseUrl) {
+            // fail silently when creating events, should not stop user interaction as echo events are collected on a best-endeavours basis
+            $this->getLogger()->warning('Echo server is not defined (missing ECHO_HOST define), not sending events');
+            return false;
+        }
+
+        $eventUrl = $baseUrl . '/events';
+
+        try {
+            $client = $this->getHttpClient();
+            $request = $client->post($eventUrl, $this->getHeaders(), $eventsData, array('connect_timeout' => 2));
+            $response = $request->send();
+
+            if ($response->isSuccessful()) {
+                $this->getLogger()->debug('Success sending events to echo');
+                return true;
+            } else {
+                $this->getLogger()->warning('Failed sending events to echo', array(
+                    'responseCode' => $response->getStatusCode(),
+                    'responseBody' => $response->getBody(true),
+                    'requestProperties' => $props
+                ));
+                return false;
+            }
+        } catch (\Exception $e) {
+            // For any exception issue, just log the issue and fail silently.  E.g. failure to connect to echo server, or whatever.
```

Comment:
As per my comments above, I'd question whether this is a decision we should make in the client, or allow the calling code to decide (i.e. by throwing exceptions).  I'm happy to discuss the relative merits of this.

I'd prefer us to throw exceptions so the client can find out about problems and choose how to respond.  We obviously need to maintain backwards compatibility, so I'd suggest we move the silent-fail case to the pre-existing `createEvent` method, in some way.

Reply:
done